apps/notifications/src/alert/services/deployment/deployment.service.ts

```diff
@@ -31,40 +35,77 @@ export class DeploymentService {
   async getDeploymentBalance(
     owner: string,
     dseq: string,
-  ): Promise<{ balance: number } | null> {
-    const { data } = await this.blockchainClientService.get<DeploymentInfo>(
-      '/akash/deployment/v1beta3/deployments/info',
-      {
-        params: {
-          'id.owner': owner,
-          'id.dseq': dseq,
-        },
-      },
-    );
+  ): Promise<Result<{ balance: number }, RichError>> {
+    try {
+      const result = await this.getDeploymentInfo(owner, dseq);
 
-    if (data.deployment.state === 'closed') {
-      this.loggerService.warn({
-        event: 'DEPLOYMENT_CLOSED',
-        owner,
-        dseq,
-        data,
-      });
+      if (result.err) {
+        return result;
+      }
 
-      return null;
-    }
+      const {
+        val: { data },
+      } = result;
+
+      if (data.deployment.state === 'closed') {
+        this.loggerService.warn({
+          event: 'DEPLOYMENT_CLOSED',
+          owner,
+          dseq,
+          data,
+        });
+
+        return Err(new RichError('Deployment closed', 'DEPLOYMENT_CLOSED'));
+      }
 
-    const balance = data.escrow_account.balance;
-    const funds = data.escrow_account.funds;
+      const balance = data.escrow_account.balance;
+      const funds = data.escrow_account.funds;
 
-    const balanceAmount =
-      balance.denom === 'uakt'
-        ? parseInt(balance.amount, 10)
-        : parseInt(balance.amount, 10) * 1_000_000;
-    const fundsAmount =
-      funds.denom === 'uakt'
-        ? parseInt(funds.amount, 10)
-        : parseInt(funds.amount, 10) * 1_000_000;
+      const balanceAmount =
+        balance.denom === 'uakt'
+          ? parseInt(balance.amount, 10)
+          : parseInt(balance.amount, 10) * 1_000_000;
+      const fundsAmount =
+        funds.denom === 'uakt'
+          ? parseInt(funds.amount, 10)
+          : parseInt(funds.amount, 10) * 1_000_000;
```

Comment:
question (blocking): if it's not `uakt`, it's probably ibc usdc, but I don't think we have to multiply by `1_000_000` as it's already `uusdc`? I'm not entirely sure

Reply:
I haven't checked honestly, took it from [here](https://github.com/akash-network/console/blob/bd79006abff3ee2d06657269ddd0e76d1554f275/apps/indexer/src/monitors/deploymentBalanceMonitor.ts#L51)

Reply:
Sure, i was going to after ur 1st comment